### PR TITLE
Add GoogleTestApp + googletest as 3rd party lib

### PR DESCRIPTION
**Pull request by NansPellicari**

### CODE REVIEW

File: GoogleTestApp.uproject [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+	"FileVersion": 3,
+	"EngineAssociation": "",
+	"Category": "",
+	"Description": "",
+	"Modules": [
+		{
+			"Name": "GoogleTest",
```
Comment (NansPellicari) [inline GoogleTestApp.uproject]: Should not be "GoogleTestApp" instead?

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "Source/ThirdParty/GoogleTest/googletest-release-1.10.0"]
```
Comment (NansPellicari) [inline .gitmodules:1]: I prefer doing this rather commit all google tests files to have a lighter project

File: GoogleTestApp.uproject (hunk 1 above)
Comment (ericlemes) [reply GoogleTestApp.uproject]: This should be GoogleTest. What we are doing here is telling that GoogleTestApp module (a console application that links to GoogleTest libraries) depends on GoogleTest module, which is the one we will include the 3rd party libraries.

**Comment by ericlemes** [comment]
The error you are having here:

c:\Users\nansp\Documents\Unreal Projects\UE4WithGoogle\Source\UE4WithGoogle.Target.cs(11,9) : error CS0103: Le nom 'DefaultBuildSettings' n'existe pas dans le contexte actuel

Is a C# error. When you run Generate Project files, Unreal Build Tool will build the .cs files (The .Build.cs and .Target.cs) in a .dll, dynamically load it and then build your C++ code, invoking their own internal magic (like Unreal Header Tool) and their own makefile syntax.

So, what is going on there is that your .Target.cs file is setting some property that doesn't exist. My bet here is that the example was built on a different Unreal Engine version than yours.

**Comment by ericlemes** [comment]
This can be of great help: https://ericlemes.com/2018/11/23/understanding-unreal-build-tool/

**Pull request merged**